### PR TITLE
feat: migrate gateway to Anchor

### PR DESCRIPTION
**Pull request by MakisChristou**
This PR introduces `axelar-solana-gateway-v2`, an Axelar gateway implementation built with the Anchor framework that maintains backwards compatibility.

### Key Changes from V1
- **Removed payload PDAs entirely** - eliminates `initialize_message_payload`, `write_message_payload`, `commit_message_payload`, and `close_message_payload` instructions
- **Effective payload size is now 1200 bytes**  as its passed in instruction data
- No Off-chain payload handling 
- PDAs and instructions remain backwards compatible with v1
- Same operator logic implementation (no operators program ) 

### Known Issues
- **Rotate signers bug (C8)** inherited from v1 

### Testing
- Focused on happy path scenarios with minimal negative test coverage
- Additional edge case testing can be added in future iterations

### CODE REVIEW

File: programs/axelar-solana-gateway-v2/src/state/message_approval.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    /// The destination address on the target blockchain.
+    pub destination_address: String,
+
+    /// A 32-byte hash of the message payload, ensuring data integrity.
+    pub payload_hash: [u8; 32],
+}
+
+impl Message {
+    // placeholder hash function
```
[the remark this answers is not in the document]
Comment (MakisChristou) [reply programs/axelar-solana-gateway-v2/src/state/message_approval.rs:73]: This is a misleading comment. The function is needed

File: programs/axelar-solana-gateway-v2/src/state/signature_verification.rs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        ) {
+            return err!(GatewayError::SignatureVerificationFailed);
+        }
+
+        let merkle_proof =
+            rs_merkle::MerkleProof::<SolanaSyscallHasher>::from_bytes(&verifier_info.merkle_proof)
+                .map_err(|_err| GatewayError::InvalidMerkleProof)?;
+
+        // simple keccak hash for now
+        let leaf_hash = verifier_info.leaf.hash();
```
[the remark this answers is not in the document]
Comment (MakisChristou) [reply programs/axelar-solana-gateway-v2/src/state/signature_verification.rs:49]: keccak is fine here the comment is misleading. What is missing here is using Digestible instead of Borsh

File: programs/axelar-solana-gateway-v2/src/state/message_approval.rs [hunk 3]
```diff
@@ -0,0 +1,115 @@
+use anchor_lang::prelude::*;
```
[the remark this answers is not in the document]
Comment (MakisChristou) [reply programs/axelar-solana-gateway-v2/src/state/message_approval.rs:1]: Not sure if zero copy is needed in general for backwards compatibility.
